### PR TITLE
Generic typenames output "fix"

### PR DESCRIPTION
**Pull request by NotOfficer**
Im writing a parser and this change makes things alot easier.
Example:
`CUtlVector< CStrongHandle< InfoForResourceTypeCTextureBase > >`
changes to
`CUtlVector<CStrongHandle<InfoForResourceTypeCTextureBase>>`

### CODE REVIEW

File: include/tools/codegen.h [hunk 1]
```diff
@@ -204,8 +204,19 @@ namespace codegen {
             return push_line(std::format("// {}", text), move_cursor_to_next_line);
         }
 
+        static void replace_all(std::string& text, const std::string& to_remove, const std::string& to_insert) {
```
Comment (es3n1n) [inline include/tools/codegen.h:207]: Not sure if we want to keep this function in the codegen class.

File: include/tools/codegen.h [hunk 2]
```diff
@@ -204,8 +204,19 @@ namespace codegen {
...
+            while (pos != std::string::npos) {
+                text.replace(pos, to_remove.length(), to_insert);
+                pos = text.find(to_remove, pos);
+            }
+        }
+
         self_ref prop(const std::string& type_name, const std::string& name, bool move_cursor_to_next_line = true) {
-            const auto line = move_cursor_to_next_line ? std::format("{} {};", type_name, name) : std::format("{} {}; ", type_name, name);
+            auto type_name_fixed = type_name;
+            replace_all(type_name_fixed, "< ", "<");
```
Comment (es3n1n) [inline include/tools/codegen.h:217]: Please don't hardcode it, add a map of replace-rules at least 😅

**Review by es3n1n — CHANGES_REQUESTED** [review]

File: include/tools/codegen.h (hunk 1 above)
Comment (NotOfficer) [reply include/tools/codegen.h]: understandable :)

File: include/tools/codegen.h (hunk 2 above)
Comment (NotOfficer) [reply include/tools/codegen.h]: i hope my solution is fine for u

**Comment by es3n1n** [comment]
Ok i thought about it a bit more and the only question that i have is that is there any other types that would have spaces in their names? Can't we just remove all the spaces?

**Comment by NotOfficer** [comment]
For example `[[maybe_unused]] uint8_t __pad0000[0x8]; // 0x0` will be affected if we remove all spaces -> `[[maybe_unused]]uint8_t ...`

**Comment by es3n1n** [comment]
> For example `[[maybe_unused]] uint8_t __pad0000[0x8]; // 0x0` will be affected if we remove all spaces -> `[[maybe_unused]]uint8_t ...`

what? no. there's no `[[maybe_unused]]` in their schema, we add it by ourselves huh

**Comment by NotOfficer** [comment]
well i just mentioned it will get affected aswell. the `[[maybe_unused]] ` part is already added when the function where i added the replacing gets called

**Comment by es3n1n** [comment]
> well i just mentioned it will get affected aswell. the `[[maybe_unused]] ` part is already added when the function where i added the replacing gets called

What stops you from replacing the spaces **before** adding the maybe_unused attribute?

**Comment by NotOfficer** [comment]
thats the first function i found to do the replacing in, the code wasnt easy to understand and i do t know for certain if its "safe" to just replace spaces for all types

**Comment by es3n1n** [comment]
I'll leave this PR open for a while. I'll get back to it whenever I have time (no ETA, sorry)

**Pull request closed without merging**